eclass/linux-mod-r1.eclass

```diff
@@ -545,10 +557,21 @@ modules_post_process() {
 	(( ${#mods[@]} )) ||
 		die "${FUNCNAME[0]} was called with no installed modules under ${path}"
 
-	# TODO?: find way for sane use with dracut (its 90kernel-modules-extra
-	# parses depmod.d files directly and assumes should include its modules
-	# which can lead to unnecessarily increased size or stale modules)
-#	_modules_process_depmod.d "${mods[@]#"${path}/"}"
+	# Instruct Dracut on whether or not we want these modules in initramfs
+	insinto /etc/dracut.conf.d
+	local dracut_arg base_name module_names=()
+	if use initramfs; then
+		dracut_arg=add_drivers
+	else
+		dracut_arg=omit_drivers
+	fi
+	for mod in "${mods[@]}"; do
+		base_name=${mod##*/}
+		module_names+=( ${base_name%.ko} )
+	done
+	newins - 10-${PN}.conf <<-EOF || die
+		${dracut_arg}="${module_names[@]}"
+	EOF
```

Comment:
I'd rather this be split in its own documented function, aka `_modules_process_dracut.conf.d "{mods[@]}"` or something.

Reply:
Done